src/query.c

```diff
@@ -498,6 +498,35 @@ IndexIterator *Query_EvalTokenNode(QueryEvalCtx *q, QueryNode *qn) {
   return NewReadIterator(ir);
 }
 
+static void addTerm(char *str, size_t tok_len, QueryEvalCtx *q, IndexIterator ***its, size_t *itsSz, size_t *itsCap, QueryNodeOptions *opts) {
+  // Create a token for the reader
+  RSToken tok = (RSToken){
+      .expanded = 0,
+      .flags = 0,
+      .len = tok_len,
+      .str = str
+  };
+
+  RSQueryTerm *term = NewQueryTerm(&tok, q->tokenId++);
+
+  // Open an index reader
+  IndexReader *ir = Redis_OpenReader(q->sctx, term, &q->sctx->spec->docs, 0,
+                                     q->opts->fieldmask & opts->fieldMask, q->conc, 1);
+
+  rm_free(tok.str);
```

Comment:
should this happen if the string is an empty string?
Maybe you can use the length in and if to make the freeing conditional.

Reply:
Yes.
It happens once so I prefer to avoid the branch in `addTerm`.

Reply:
So you prefer to do a memory allocation every time over an if?

Reply:
Yes
It's once per fuzzy query only when the empty string is relevant, while the `if` will be executed for every expanded term from the base term, which may be a lot!

Reply:
Maybe move the `rm_free` outside to the caller?
Then you will only need to call it in the while loop.

Reply:
Also, the less `if`s the better :)
Especially when the price is so low.

Reply:
Moving the `rm_free` outside removes the need for an if and removes the need for the strdup